### PR TITLE
DAO468 depositAndDelegate function

### PR DESCRIPTION
**Pull request by cybervoid0**
# What 

- in `stRIF` smart contract create a function called `depositAndDelegate` 
- write test cases for happy \ unhappy paths
- all tests are passing
- slither doesnt find any issues

# Why

This function will allow user to stake and delegate within a single transaction

# Ref

[DAO-468](https://rsklabs.atlassian.net/browse/DAO-468)

### CODE REVIEW

**Review by rodrigoncalves — APPROVED** [review]
Nice job

File: test/StRIFToken.depositAndDelegate.test.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  let stRifAddress: Address
+  let mintDelegateTx: ContractTransactionResponse
+  const votingPower = 100n * 10n ** 18n // 100 RIF tokens
+
+  before(async () => {
+    /* 
+    Deployment of RIF token and transferring a certain amount to a holder.
+    See the relevant cases in StRIFToken tests
+    */
+    ;[deployer, holder, stranger] = await ethers.getSigners()
```
Comment (sleyter93) [inline test/StRIFToken.depositAndDelegate.test.ts:26]: I think we can remove the ";"

File: test/StRIFToken.depositAndDelegate.test.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    rif = await (await ethers.deployContract('RIFToken')).waitForDeployment()
+    rifAddress = (await rif.getAddress()) as Address
+    await (await rif.setAuthorizedManagerContract(deployer.address)).wait()
+    const latestBlock = await ethers.provider.getBlock('latest')
+    if (!latestBlock) throw new Error('Latest block not found')
+    await (await rif.closeTokenDistribution(latestBlock.timestamp)).wait()
+    stRif = await deployStRIF(rifAddress, deployer.address)
```
Comment (sleyter93) [inline test/StRIFToken.depositAndDelegate.test.ts:33]: Maybe we can add some comments like:
// Deploying and Setting Token RIF
// Deploying StRIF
// Transferring initial funds

File: test/StRIFToken.depositAndDelegate.test.ts (hunk 1 above)
Comment (cybervoid0) [reply test/StRIFToken.depositAndDelegate.test.ts:26]: this semicolon is needed and was put automatically by prettier

**Pull request closed without merging**